### PR TITLE
Audio device select

### PR DESCRIPTION
**Pull request by psmokotnin**
IAudioDriver: added methods for getting audio device list, current device and select device.
macOS: implemented these methods

### CODE REVIEW

File: src/framework/audio/internal/iaudiodriver.h [hunk 1]
```diff
@@ -55,6 +57,11 @@ class IAudioDriver : MODULE_EXPORT_INTERFACE
     virtual bool open(const Spec& spec, Spec* activeSpec) = 0;
     virtual void close() = 0;
     virtual bool isOpened() const = 0;
+
+    virtual std::string device() const = 0;
+    virtual bool selectDevice(std::string name) = 0;
+    virtual std::vector<std::string> availableDevices() const = 0;
+    virtual async::Notification deviceListChanged() const = 0;
```
Comment (igorkorsukov) [inline src/framework/audio/internal/iaudiodriver.h:64]: Slightly inconsistent names. It is better that the names are corresponding -  `someThing()` and `someThingChanged()`
i.e. `availableDevices()` and `availableDevicesChanged()` or `deviceList()` and `deviceListChanged()`

File: src/framework/audio/internal/platform/osx/osxaudiodriver.cpp [hunk 2]
```diff
@@ -123,6 +131,120 @@ bool OSXAudioDriver::isOpened() const
...
+        propertySize = CFStringGetLength(nameRef);
+        deviceName.resize(propertySize);
+        CFStringGetCString(nameRef, deviceName.data(), sizeof(deviceName), kCFStringEncodingUTF8);
+        m_devices[d] = deviceName;
+        CFRelease(nameRef);
+    }
+    m_deviceListChanged.notify();
+}
+
+bool OSXAudioDriver::audioQueueSetDeviceName(std::string deviceName)
```
Comment (igorkorsukov) [inline src/framework/audio/internal/platform/osx/osxaudiodriver.cpp:200]: better use `const td::string&`

File: src/framework/audio/internal/iaudiodriver.h [hunk 3]
```diff
@@ -55,6 +57,11 @@ class IAudioDriver : MODULE_EXPORT_INTERFACE
     virtual bool open(const Spec& spec, Spec* activeSpec) = 0;
     virtual void close() = 0;
     virtual bool isOpened() const = 0;
+
+    virtual std::string device() const = 0;
+    virtual bool selectDevice(std::string name) = 0;
```
Comment (igorkorsukov) [inline src/framework/audio/internal/iaudiodriver.h:62]: better use `const td::string&`

Comment (psmokotnin) [reply src/framework/audio/internal/iaudiodriver.h]: fixed

File: src/framework/audio/internal/platform/osx/osxaudiodriver.cpp (hunk 2 above)
Comment (psmokotnin) [reply src/framework/audio/internal/platform/osx/osxaudiodriver.cpp]: fixed

File: src/framework/audio/internal/iaudiodriver.h (hunk 1 above)
Comment (psmokotnin) [reply src/framework/audio/internal/iaudiodriver.h]: fixed

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**